src/backend/database_models/agent_tool_metadata.py

```diff
@@ -0,0 +1,27 @@
+from typing import Optional
+
+from sqlalchemy import ForeignKey, Text, UniqueConstraint
+from sqlalchemy.dialects.postgresql import ARRAY
+from sqlalchemy.orm import Mapped, mapped_column
+
+from backend.database_models.base import Base
+
+
+class AgentToolMetadata(Base):
+    __tablename__ = "agent_tool_metadata"
+
+    user_id: Mapped[str] = mapped_column(Text, nullable=False)
+    agent_id: Mapped[str] = mapped_column(
+        ForeignKey("agents.id", ondelete="CASCADE"), nullable=False
+    )
+    tool_name: Mapped[str] = mapped_column(Text, nullable=False)
+    google_drive_file_ids: Mapped[list[str]] = mapped_column(ARRAY(Text), nullable=True)
+    google_drive_folder_ids: Mapped[list[str]] = mapped_column(
```

Comment:
@giannis2two i noticed the gdrive tool differentiates folder id and file id. so i decided the easiest way is to just have two columns here.

Reply:
i realized that might be a bit tough to generalize a column to all tools since we kinda need granular separation between tools and what they want out of these metadata entries.

for example, if we just had one column as artifacts we cant differentiate between folder and file ids. also it would rely on the developer to know exactly what the artifacts hold.

Reply:
giannis suggested just having a type column and artifact column, however, that leads back to this q https://github.com/cohere-ai/cohere-toolkit/pull/309/files#r1657610964

Reply:
resolved